specs/testing/SkillsMatrixTesting.md

```diff
@@ -0,0 +1,58 @@
+# Automated Testing of Skills
+Defines the short term goals around the automated testing of skills. 
+
+# Goal
+Automate the testing matrix for Bot/Skill interactions. The full testing matrix runs every night. Failures result in emails. 
+
+# Version of the Botbuilder SDK
+1. For each test run, the latest *preview* Bot Builder SDK should be consumed from the MyGet.
+
+2. Alternatly, a specific version of the BotBuilder SDK could be specified via an Azure Devops variable. 
+
+# Testing Matrix
+The following testing matrix needs to be fully tested
+|   |        |    | **Skill**    |        |   |
+|---|--------|----|------------|--------|---|
+|   |        | C# | Javascript | Python |   |
+|   | C#     | 1  |     2      | 3      |   |
+|   | JS     | 4  |5           | 6      |   |
+|   | Python | 7  |8           | 9      |   |
+
+1. C# Bot acting as consumer, calling a C# Skill
+2. C# Bot acting as consumer, calling a JS Skill
+3. C# Bot acting as consumer, calling a Pyton Skill
+4. JS Bot acting as consumer, calling a C# Skill
+5. JS Bot acting as consumer, calling a JS Skill
+6. JS Bot acting as consumer, calling a Python Skill
+7. Python Bot acting as consumer, calling a C# Skill
+8. Python Bot acting as consumer, calling a JS Skill
+9. Python Bot acting as consumer, calling a Python Skill
+
+# Tests
+V0 of the tests is an Echo. The Consumer bot will send a Message containing a GUID to the skill, and the skill encho that back. 
```

Comment:
I think the tests should also include sending an "end" utterance that will trigger returning the EndOfConversation activity from the skill (and processed from the consumer).